lib/ex_doc/cli.ex

```diff
@@ -35,6 +36,11 @@ defmodule ExDoc.CLI do
     else
       generate(args, opts, generator)
     end
+
+    if List.keymember?(opts, :warnings_as_errors, 0) and ExDoc.WarningCounter.count() > 0 do
```

Comment:
we should check for the value:

```suggestion
    if opts[:warnings_as_errors] == true and ExDoc.WarningCounter.count() > 0 do
```

because, though extremely unlikely, someone could set it to false and then it wouldn't work:

```elixir
iex> OptionParser.parse!(~w(--warnings-as-errors=false), strict: [warnings_as_errors: :boolean])
{[warnings_as_errors: false], []}
```